src/stream_io.rs

```diff
@@ -0,0 +1,84 @@
+use std::fmt::Arguments;
+use std::io;
+use std::io::{Read, Write, Seek, SeekFrom};
+use std::os::raw::{c_char, c_int, c_void};
+use std::slice;
+
+/// Returns 0 on success and 1 on failure.
+///
+/// ImfIO.h:
+/// virtual void write (const char c[/*n*/], int n) = 0;
+pub unsafe extern "C" fn write_stream<T: Write>(writer: *mut c_void,
+                                                c: *const c_char,
+                                                n: c_int)
+                                                -> c_int {
+    let bytes = slice::from_raw_parts(c as *const u8, n as usize);
+    if let Ok(_) = (*(writer as *mut T)).write_all(bytes) {
+        return 0;
+    } else {
+        return 1;
+    }
+}
+
+/// Returns 0 on success and 1 on failure.
+///
+/// ImfIO.h:
+/// virtual void seekp (Int64 pos) = 0;
+pub unsafe extern "C" fn seek_stream<T: Seek>(seeker: *mut c_void, pos: u64) -> c_int {
+    if let Ok(_) = (*(seeker as *mut T)).seek(SeekFrom::Start(pos)) {
+        return 0;
+    } else {
+        return 1;
+    }
+}
+
+// ----------------------------------------------------------------
+
+// Indicates an unused io stream in the type parameters of the various
+// input/output file types.
+pub struct Unused {}
```

Comment:
I'm happy removing that, yeah.  It also reduces the FFI surface area to take it out, which is nice.  I'll get rid of it.